### PR TITLE
Improve basic QC plots

### PR DESCRIPTION
**Pull request by bbimber**

### CODE REVIEW

File: R/BFF_Demux.R [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 utils::globalVariables(
   names = c('relative_counts', 'x', 'y', '..density..', 'Highest', 'Second'),
-  package = 'cellhashR',
+  # package = 'cellhashR',
```
Comment (bbimber) [inline R/BFF_Demux.R:7]: should restore this?

File: R/BFF_Demux.R [hunk 2]
```diff
@@ -90,15 +90,23 @@ PlotCutoff <- function(data, smooth, label) {
...
-    ggtitle(paste0("Histogram of ", label)) +
-    xlab("Log(HTO Counts)") +
-    geom_line(data = data.frame(x = smooth$x, y = sqrt(smooth$y)), mapping = aes(x = x, y = y), color = "blue", size = 1) 
+  
+  
+  plotdata <- data.frame(Value = data)
+  linedata <- data.frame(x = smooth$x, y = sqrt(smooth$y))
+  
+  
+  # P1 <- ggplot(data.frame(Value = data), aes(x = Value)) +
```
Comment (bbimber) [inline R/BFF_Demux.R:99]: should remove old code if we dont need it

File: R/BFF_Demux.R [hunk 3]
```diff
@@ -90,15 +90,23 @@ PlotCutoff <- function(data, smooth, label) {
...
+  #   ggtitle(label) + ylab("") + 
+  #   xlab("") +
+  #   geom_line(data = data.frame(x = smooth$x, y = sqrt(smooth$y)), mapping = aes(x = x, y = y), color = "blue", size = 1) +
+  #   egg::theme_presentation(base_size = 8)
+  
+  
   # + geom_line(data = data.frame(x = smooth$x, y = 50*deriv), mapping = aes(x = x, y = y), color = "red", size = 1)
   
-  ymax <- max(graphics::hist(data, breaks = nbins, plot=FALSE)$density)
+  # ymax <- max(graphics::hist(data, breaks = nbins, plot=FALSE)$density)
```
Comment (bbimber) [inline R/BFF_Demux.R:109]: remove?

File: R/BFF_Demux.R [hunk 4]
```diff
@@ -111,11 +119,12 @@ PlotCutoff <- function(data, smooth, label) {
     if (x2 < x1){
       #Negative peak must be at least 1/10th the positive peak
       if (y2 < (y1/10)) {
-        print(label)
         print('Negative peak was not at least 1/10th the positive peak, using max value as cutoff')
-        P1 <- P1 + geom_vline(xintercept = max(data), size = 1)
-        print(P1)
-        return(list(max(data), P1))
+        # P1 <- P1 + geom_vline(xintercept = max(data), size = 1)
```
Comment (bbimber) [inline R/BFF_Demux.R:123]: remove?

File: R/BFF_Demux.R [hunk 5]
```diff
@@ -124,15 +133,16 @@ PlotCutoff <- function(data, smooth, label) {
     cutoff_index <- which.min(smooth$y[cutoff_indices]) + min(cutoff_indices)
     cutoff <- smooth$x[cutoff_index]
   } else {
-    print(label)
     print('Only one peak found, using max value as cutoff')
-    P1 <- P1 + geom_vline(xintercept = max(data), size = 1)
-    print(P1)
-    return(list(max(data), P1))
+    # P1 <- P1 + geom_vline(xintercept = max(data), size = 1)
```
Comment (bbimber) [inline R/BFF_Demux.R:137]: remove?

File: R/BFF_Demux.R [hunk 6]
```diff
@@ -264,27 +374,41 @@ BFFDemux <- function(seuratObj, assay, simple_threshold=simple_threshold, double
 
   #loop over HTOs in matrix, perform thresholding and store cells that pass the threshold
   #return a discrete matrix, with 1 equal to a positive call for that barcode
-  discrete <- GetAssayData(object = seuratObj, assay = assay)
-  discrete[discrete > 0] <- 0
+
```
Comment (bbimber) [inline R/BFF_Demux.R:377]: remove?

File: R/BFF_Demux.R [hunk 7]
```diff
@@ -299,21 +423,29 @@ BFFDemux <- function(seuratObj, assay, simple_threshold=simple_threshold, double
...
-        print("Threshold may be placed too low, recalculating with more smoothing.")
-        cutoffresults <- getCountCutoff(cells, hto, 2)
-      }
-      threshold <- (cutoffresults[[1]])
-      max_list <- rbind(max_list, cutoffresults[[3]])
-      normed_cutoffs[[hto]] <- threshold
-    }
+    normedres <- generateBFFGridPlot(t(tot_normed), barcodeBlocklist= NULL, "Log(Counts + 1)", "Normalized Count Distributions with Fitted Threshold")
     
+    # normed_cutoffs <- list()
```
Comment (bbimber) [inline R/BFF_Demux.R:428]: remove?

File: R/Normalization.R [hunk 8]
```diff
@@ -207,4 +235,32 @@ PerformHashingClustering <- function(barcodeMatrix, norm) {
...
+    label <- paste0(norm, ' Normalized Counts')
+    maintitle <- ggplot2::ggtitle(paste0(norm, " Normalized HTO Barcode Count Distributions"))
+  }
+
+  df <- data.frame(df, check.names=FALSE)
+  df <- tibble::rownames_to_column(df, var = "cell")
+  df <- df %>% tidyr::pivot_longer(colnames(df)[2:length(colnames(df))], names_to = "Barcode", values_to = "count")
+  P1 <- df %>%
+    dplyr::mutate(Barcode = factor(Barcode, levels=unique(df$Barcode)))  %>%
+    ggplot(aes( y=count, x=Barcode)) + # stat_summary(aes(y = sqrt(..density..)), colour = "red", size = 2, geom="violin") +
```
Comment (bbimber) [inline R/Normalization.R:255]: remove commented code?

File: R/BFF_Demux.R (hunk 1 above)
Comment (gboggy) [reply R/BFF_Demux.R:7]: yes.  Thanks!

**Pull request merged**